pkg/services/oauth/oauth.go

```diff
@@ -95,28 +118,19 @@ func (s *OAuthService) ParseAndValidateJwtBearerToken(acString string) (*service
 func (s *OAuthService) ParseAndValidateAccessToken(accessToken string) (*services.NutsAccessToken, error) {
 	parser := &jwt.Parser{ValidMethods: []string{jwt.SigningMethodRS256.Name}}
 	token, err := parser.ParseWithClaims(accessToken, &services.NutsAccessToken{}, func(token *jwt.Token) (i interface{}, e error) {
-		legalEntity, err := parseTokenIssuer(token.Claims.(*services.NutsAccessToken).Issuer)
-		if err != nil {
-			return nil, err
-		}
-
 		// Check if the care provider which signed the token is managed by this node
-		if !s.Crypto.PrivateKeyExists(nutsCryptoTypes.KeyForEntity(nutsCryptoTypes.LegalEntity{URI: legalEntity.String()})) {
-			return nil, fmt.Errorf("invalid token: not signed by a care provider of this node")
+		if !s.Crypto.PrivateKeyExists(s.oauthKeyEntity) {
+			return nil, fmt.Errorf("invalid token: not signed by this node")
```

Comment:
I think this error message is wrong. It's more like "service not configured" or "OAuth JWT private key missing"

Reply:
If we really want to be able to give this error message; did we decide to put the x5c field in the JWT? In that case you could compare the public key with the one we have stored. Otherwise "invalid signature" indicating the wrong key was used is probably fine, too

Reply:
went with the last one. This method will be changed anyway to fully support RFC003